src/sources/audiosource.h

```diff
@@ -3,235 +3,197 @@
 
 #include "sources/urlresource.h"
 #include "util/audiosignal.h"
+#include "util/indexrange.h"
 #include "util/memory.h"
 #include "util/result.h"
 #include "util/samplebuffer.h"
 
+
 namespace mixxx {
 
 // forward declaration(s)
+class AudioSource;
 class AudioSourceConfig;
 
+typedef std::shared_ptr<AudioSource> AudioSourcePointer;
+
+// AudioSource v2 interface
+class IAudioSource {
+  public:
+    virtual ~IAudioSource() = default;
+
+    // Read sample frames from the requested frame index range. Only
+    // forward-oriented index ranges need to be supported. The implementation
+    // is responsible for restricting the requested range to the range that
+    // is actually readable.
+    //
+    // Only that part of the output buffer corresponding to the returned
+    // range is allowed to be modified. All remaining samples in the output
+    // buffer should stay untouched. The samples in the output buffer need
+    // to be offset properly depending on the difference between the first
+    // requested frame and the actual first read frame, i.e. the first sample
+    // in the output buffer represents the sample of the first channel from
+    // the first requested frame.
+    //
+    // If the output buffer is null all decoded samples must be discarded
+    // immediately. Otherwise the size of the output buffer must be
+    // sufficient to buffer all frames2samples(frameIndexRange.length())
+    // sample values.
+    //
+    // On errors only a sub range of the requested frames might be returned.
+    virtual IndexRange readOrSkipSampleFrames(
```

Comment:
After reading the code I understand the function name. But the "OrSkip" part was confusing on the first read. 
Now it is obvious that if you pass a null buffer that it skips frames. Isn't the main nature of this function that it seeks to the file and fills the buffer if it can do it? 

How about seekAndReadSampleFrames()